src/engine/TransitivePath.cpp

```diff
@@ -658,119 +449,158 @@ void TransitivePath::computeTransitivePathRightBound(
+                                 const TransitivePathSide& targetSide) const {
+  std::vector<Id> nodes;
+  Map edges = setupEdgesMap<SUB_WIDTH>(sub, startSide, targetSide);
+
+  // id -> var|id
+  if (startSide.isBound()) {
+    nodes.push_back(std::get<Id>(startSide.value));
+    // var -> var
   } else {
-    p->_rightSideTree = leftOrRightOp;
-    p->_rightSideCol = inputCol;
+    nodes = setupNodesVector<SUB_WIDTH>(sub, startSide.subCol);
+    if (_minDist == 0) {
+      std::vector<Id> targetNodes =
+          setupNodesVector<SUB_WIDTH>(sub, targetSide.subCol);
+      nodes.insert(nodes.end(), targetNodes.begin(), targetNodes.end());
+    }
   }
 
-  // Note: The `variable` in the following structured binding is `const`, even
-  // if we bind by value. We deliberately make one unnecessary copy of the
-  // `variable` to keep the code simpler.
-  for (auto [variable, columnIndexWithType] :
-       leftOrRightOp->getVariableColumns()) {
-    ColumnIndex columnIndex = columnIndexWithType.columnIndex_;
-    if (columnIndex != inputCol) {
-      if (columnIndex > inputCol) {
-        columnIndexWithType.columnIndex_++;
-        p->_variableColumns[variable] = columnIndexWithType;
-      } else {
-        columnIndexWithType.columnIndex_ += 2;
-        p->_variableColumns[variable] = columnIndexWithType;
-      }
-      p->_resultWidth++;
+  return std::make_pair(edges, nodes);
+}
+
+// _____________________________________________________________________________
+template <size_t SUB_WIDTH>
+TransitivePath::Map TransitivePath::setupEdgesMap(
+    const IdTable& dynSub, const TransitivePathSide& startSide,
+    const TransitivePathSide& targetSide) {
+  const IdTableView<SUB_WIDTH> sub = dynSub.asStaticView<SUB_WIDTH>();
+  Map edges;
+
+  for (size_t i = 0; i < sub.size(); i++) {
+    // checkTimeoutHashSet();
+    Id startId = sub(i, startSide.subCol);
+    Id targetId = sub(i, targetSide.subCol);
+    MapIt it = edges.find(startId);
+    if (it == edges.end()) {
+      std::shared_ptr<ad_utility::HashSet<Id>> edgeTargets =
+          std::make_shared<ad_utility::HashSet<Id>>();
+      edgeTargets->insert(targetId);
+      edges[startId] = edgeTargets;
+    } else {
+      // If r is not in the vector insert it
+      it->second->insert(targetId);
     }
   }
-  return p;
+  return edges;
 }
 
 // _____________________________________________________________________________
-bool TransitivePath::isBound() const {
-  return _leftSideTree != nullptr || _rightSideTree != nullptr || !_leftIsVar ||
-         !_rightIsVar;
+template <size_t WIDTH>
+std::vector<Id> TransitivePath::setupNodesVector(const IdTable& table,
+                                                 size_t col) {
+  std::vector<Id> nodes;
+  const IdTableView<WIDTH> tableView = table.asStaticView<WIDTH>();
+  for (size_t i = 0; i < tableView.size(); i++) {
+    nodes.push_back(tableView(i, col));
+  }
+  return nodes;
+}
```

Comment:
I think we don't need this anymore with the columnBased IdTable.
You can simply always use `std::span<Id>` (a lightweight view) and just return 
`table.getColumn(col)` here.
Then you get rid of the copies and allocations etc.